xtable-service/src/main/java/org/apache/xtable/service/ConversionService.java

```diff
@@ -0,0 +1,151 @@
+    this.sparkHolder = sparkHolder;
+    this.controllerFactory = controllerFactory;
+    this.icebergUtil = icebergUtil;
+    this.hudiUtil = hudiUtil;
+    this.deltaUtil = deltaUtil;
+  }
+
+  public ConvertTableResponse convertTable(ConvertTableRequest request) {
+    Configuration conf = sparkHolder.jsc().hadoopConfiguration();
+    ConversionController conversionController = controllerFactory.create(conf);
+
+    SourceTable sourceTable =
+        SourceTable.builder()
+            .name(request.getSourceTableName())
+            .basePath(request.getSourceTablePath())
+            .formatName(request.getSourceFormat())
+            .build();
+
+    List<TargetTable> targetTables = new ArrayList<>();
+    for (String targetFormat : request.getTargetFormats()) {
+      TargetTable targetTable =
+          TargetTable.builder()
+              .name(request.getSourceTableName())
+              .basePath(request.getSourceTablePath())
+              .formatName(targetFormat)
+              .build();
+      targetTables.add(targetTable);
+    }
+    ConversionConfig conversionConfig =
+        ConversionConfig.builder().sourceTable(sourceTable).targetTables(targetTables).build();
+    ConversionSourceProvider<?> conversionSourceProvider =
+        getConversionSourceProvider(request.getSourceFormat());
+    conversionController.sync(conversionConfig, conversionSourceProvider);
+
+    List<RestTargetTable> restTargetTables = new ArrayList<>();
+    for (String targetFormat : request.getTargetFormats()) {
+      if (targetFormat.equals("ICEBERG")) {
+        Pair<String, String> responseFields =
+                icebergUtil.getIcebergSchemaAndMetadataPath(
+                        request.getSourceTablePath(), sparkHolder.jsc().hadoopConfiguration());
+        RestTargetTable icebergTable =
+                new RestTargetTable("ICEBERG", responseFields.getLeft(), responseFields.getRight());
+        restTargetTables.add(icebergTable);
+      } else if (targetFormat.equals("HUDI")) {
+        Pair<String, String> responseFields =
+                hudiUtil.getHudiSchemaAndMetadataPath(
+                        request.getSourceTablePath(), sparkHolder.jsc().hadoopConfiguration());
+        RestTargetTable hudiTable =
+                new RestTargetTable("HUDI", responseFields.getLeft(), responseFields.getRight());
+        restTargetTables.add(hudiTable);
+      } else if(targetFormat.equals("DELTA")){
+        Pair<String, String> responseFields =
+                deltaUtil.getDeltaSchemaAndMetadataPath(
+                        request.getSourceTablePath(), sparkHolder.spark());
+        RestTargetTable deltaTable =
+                new RestTargetTable("DELTA", responseFields.getLeft(), responseFields.getRight());
+        restTargetTables.add(deltaTable);
+      }
+    }
+    return new ConvertTableResponse(restTargetTables);
+  }
+
+  private ConversionSourceProvider<?> getConversionSourceProvider(String sourceTableFormat) {
```

Comment:
There's a similar method in utilities, can we move this to xtable-core and import it from there? This could just be a static method taking in sourceTableFormat and a configuration object.

Reply:
will see if i can do this.

Reply:
Im actually refactoring the hudi, iceberg, delta util into one conversion service util, i think we should opt to move it in the same util class for now.